tests/phpunit/tests/html-api/wpHtmlProcessorSubclass.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Processor subclass functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML-API
+ *
+ * @since 6.6.0
+ *
+ * @group html-api
+ *
+ * @coversDefaultClass WP_HTML_Processor
+ */
+class Tests_HtmlApi_WpHtmlProcessorSubclass extends WP_UnitTestCase {
+	/**
+	 * Ensures that subclasses can be created from ::create_fragment method.
+	 *
+	 * @ticket TBD
+	 */
+	public function test_subclass_create_fragment_creates_subclass() {
+		$processor = WP_HTML_Processor::create_fragment( '' );
+		$this->assertInstanceOf( WP_HTML_Processor::class, $processor, '::create_fragment did not return class instance.' );
+
+		$processor = WP_HTML_Processor_Subclass::create_fragment( '' );
+		$this->assertInstanceOf( WP_HTML_Processor_Subclass::class, $processor, '::create_fragment did not return subclass instance.' );
+	}
+}
```

Comment:
maybe we should put this into the `wpHtmlProcessor.php` test file and create an anonymous subclass? I  don't think it deserves its own file just for this one test.

Reply:
Agreed and done.